lib/obs.rb

```diff
@@ -143,6 +144,38 @@ def initialize
   #
   def self.search_published_binary(query, opts = {})
       result = OBS.client.get('/search/published/binary/id', match: xpath_for(query, opts)).body.collection
-      result.binaries.present? result.binaries : []
+      return [] if result.binaries.nil?
+
+      result.binaries.map { |bin| OBS.add_project_quality(bin) }
+  end
+
+  # Add project quality attribute for filtering
+  # @param [Binary] to add its project's quality attribute to
+  #
+  # @return [Binary]
+  def self.add_project_quality(binary)
+    begin
+      binary.quality = OBS.search_project_quality(binary.project)
+    rescue Faraday::ClientError => e
+      # Attribute search might fail, not all projects have accessible info
+      raise unless e.response[:status] == 404
+
+      # Assume the project is NOT private if search fails
+      binary.quality = ""
+    end
+    binary
+  end
+
+  # Searches for the OBS:QualityCategory attribute of a given project
+  # @param [String] project OBS specific project to search attribute for
+  #
+  # @return [String]
+  def self.search_project_quality(project)
+    cache_key = ActiveSupport::Cache.expand_cache_key(project, "attribute")
+    Rails.cache.fetch(cache_key, expires_in: 2.hours) do
+      xml_quality = OBS.client.get("/source/#{project}/_attribute/OBS:QualityCategory")
+      quality = xml_quality[:attribute][:text].strip unless xml_quality.nil? || xml_quality[:attribute].nil?
+      quality.nil? ? "" : quality
```

Comment:
`quality ? '' : quality`

why is rubocop not complaining about your `""`? :rage:

Reply:
Even better, combine the two lines:
`xml_quality && xml_quality[:attribute] ? xml_quality[:attribute][:text].strip : ''`